docs/sources/api/_index.md

```diff
@@ -805,13 +806,31 @@ In microservices mode, the `/flush` endpoint is exposed by the ingester.
 
 ## `POST /ingester/flush_shutdown`
 
+**Deprecated**: Please use `/ingester/shutdown?flush=true` instead.
+
 `/ingester/flush_shutdown` triggers a shutdown of the ingester and notably will _always_ flush any in memory chunks it holds.
 This is helpful for scaling down WAL-enabled ingesters where we want to ensure old WAL directories are not orphaned,
 but instead flushed to our chunk backend.
 
 In microservices mode, the `/ingester/flush_shutdown` endpoint is exposed by the ingester.
 
-### `GET /distributor/ring`
+## `POST /ingester/shutdown`
+
+`/ingester/shutdown` is similar to the [`/ingester/flush_shutdown`](#post-ingesterflush_shutdown)
+endpoint, but accepts two URL query parameters `flush` and `delete_ring_tokens`.
+
+**URL query parameters:**
+
+* `flush=<bool>`:
+  Flag to control whether to flush any in-memory chunks the ingester holds. Defaults to `true`.
+* `delete_ring_tokens=<bool>`:
+  Flag to control whether to delete the file that contains the ingester ring tokens of the instance if the `-ingester.token-file-path` is specified.
+
+This handler, in contrast to the `/ingester/flush_shutdown` handler, terminates the Loki process.
```

Comment:
I'm curious why we don't expose a flag for the shutdown - can you help me understand pls?

Reply:
Not sure what you exactly mean with that.
* Adding a flag to the existing `/ingester/flush_shutdown` handler to allow the process to be terminated?
* Or having a flag on `/ingester/shutdown` to allow either stopping the service only, or terminating the process completely?

IMO, "shutdown" suggests that the process is terminated, that's why I wanted to keep the behaviour of the old handler but deprecate it, and the new handler _always_ terminates the process (which I think is almost always what you want).

Reply:
I think it's the "almost always" there that's making me hesitate. We're changing behaviour now, which some users may rely on. I think we need to maintain the ability to shutdown the ingester but not terminate; it can default to `true` if this is what we recommend.